web_poet/pages.py

```diff
@@ -34,15 +36,33 @@ def is_injectable(cls: typing.Any) -> bool:
     return isinstance(cls, type) and issubclass(cls, Injectable)
 
 
-class ItemPage(Injectable, abc.ABC):
-    """Base Page Object with a required :meth:`to_item` method.
-    Make sure you're creating Page Objects with ``to_item`` methods
-    if their main goal is to extract a single data record from a web page.
+ItemT = typing.TypeVar("ItemT")
+
+
+class ItemPage(Injectable, typing.Generic[ItemT]):
+    """Base Page Object, with a default :meth:`to_item` implementation
+    which supports web-poet fields.
     """
 
-    @abc.abstractmethod
-    def to_item(self):
+    _skip_nonitem_fields: bool
+
+    def __init_subclass__(cls, skip_nonitem_fields: bool = False, **kwargs):
+        super().__init_subclass__(**kwargs)
+        cls._skip_nonitem_fields = skip_nonitem_fields
+
+    @property
+    def item_cls(self) -> typing.Type[ItemT]:
+        """Item class"""
+        param = get_generic_parameter(self.__class__)
+        if isinstance(param, typing.TypeVar):  # class is not parametrized
+            return dict  # type: ignore[return-value]
+        return param
+
+    async def to_item(self) -> ItemT:
         """Extract an item from a web page"""
+        return await item_from_fields(
+            self, item_cls=self.item_cls, item_cls_fields=self._skip_nonitem_fields
```

Comment:
I wonder if it makes sense to rename `item_cls_fields` argument to `skip_nonitem_fields`